zkvm/src/ops.rs

```diff
@@ -202,7 +202,9 @@ impl Instruction {
         match self {
             Instruction::Push(data) => {
                 write(Opcode::Push);
-                data.encode(program);
+                let mut bytes = data.to_bytes();
+                encoding::write_u32(bytes.len() as u32, program);
```

Comment:
yeah, it's good to have it here, since it's the same module where this length prefix is parsed.